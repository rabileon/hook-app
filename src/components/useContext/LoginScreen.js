import React, { useContext } from 'react';
import { UserContext } from './UseContext';

export const LoginScreen = () => {
  //1. Obtener la referencia al userContext
  const { setUser } = useContext(UserContext);
  //2. setUser

  return (
    <div>
      <h1>LoginScreen</h1>
      <hr />
      <button
        className="btn btn-primary"
        onClick={() => setUser({ id: 123, name: 'Rabi' })}
      >
        Login
      </button>
    </div>
  );
};
