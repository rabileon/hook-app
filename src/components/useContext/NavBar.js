import React from 'react';
import { Link, NavLink } from 'react-router-dom';
export const NavBar = () => {
  return (
    <nav className="navbar navbar-expand-sm navbar-dark bg-dark">
      <div className="container-fluid">
        <NavLink to="/" className="navbar-brand">
          useContext
        </NavLink>

        <div className="collapse navbar-collapse" id="navbarNavAltMarkup">
          <div className="navbar-nav">
            <NavLink
              exact="true"
              activeclassname="active"
              to="/"
              className="nav-link"
              aria-current="page"
            >
              Home
            </NavLink>
            <NavLink
              exact="true"
              activeclassname="active"
              to="/about"
              className="nav-link"
              aria-current="page"
            >
              About
            </NavLink>
            <NavLink
              exact="true"
              activeclassname="active"
              to="/login"
              className="nav-link"
              aria-current="page"
            >
              Login
            </NavLink>
          </div>
        </div>
      </div>
    </nav>
  );
};
