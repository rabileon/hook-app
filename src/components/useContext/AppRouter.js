import React from 'react';
import {
  BrowserRouter as Router,
  Routes,
  Route,
  Navigate,
} from 'react-router-dom';
import { AboutScreen } from './AboutScreen';
import { HomeScreen } from './HomeScreen';
import { LoginScreen } from './LoginScreen';
import { NavBar } from './NavBar';
export const AppRouter = () => {
  return (
    <Router>
      <div>
        <NavBar></NavBar>
        <Routes>
          <Route exact path="/" element={<HomeScreen />} />
          <Route exac path="/about" element={<AboutScreen />} />
          <Route exac path="/login" element={<LoginScreen />} />
          <Route path="*" element={<Navigate to="/" />} />
        </Routes>
      </div>
    </Router>
  );
};
