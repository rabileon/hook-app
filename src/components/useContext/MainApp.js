import React from 'react';
import { AppRouter } from './AppRouter';
import { UserContext } from './UseContext';
export const MainApp = () => {
  const user = {
    id: 1234,
    name: 'Ferando',
    email: 'rleon@hotmail.com',
  };
  return (
    <UserContext.Provider value={user}>
      <AppRouter></AppRouter>
    </UserContext.Provider>
  );
};
