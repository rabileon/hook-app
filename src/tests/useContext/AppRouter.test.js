import React from 'react';
import { mount } from 'enzyme';
import { AppRouter } from '../../components/useContext/AppRouter';
import { UserContext } from '../../components/useContext/UseContext';

describe('Pruebas en <AppRouter />', () => {
  const user = {
    id: 123,
    name: 'Rabi',
  };

  const wrapper = mount(
    <UserContext.Provider value={{ user }}>
      <AppRouter />
    </UserContext.Provider>
  );
  test('debe de mostrarse correctamente', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
