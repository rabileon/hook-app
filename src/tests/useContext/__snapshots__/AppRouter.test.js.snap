// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AppRouter /> debe de mostrarse correctamente 1`] = `
<div>
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <div
      className="container-fluid"
    >
      <a
        aria-current="page"
        className="navbar-brand active"
        href="/"
        onClick={[Function]}
      >
        useContext
      </a>
      <div
        className="collapse navbar-collapse"
        id="navbarNavAltMarkup"
      >
        <div
          className="navbar-nav"
        >
          <a
            activeclassname="active"
            aria-current="page"
            className="nav-link active"
            exact="true"
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
          <a
            activeclassname="active"
            className="nav-link"
            exact="true"
            href="/about"
            onClick={[Function]}
          >
            About
          </a>
          <a
            activeclassname="active"
            className="nav-link"
            exact="true"
            href="/login"
            onClick={[Function]}
          >
            Login
          </a>
        </div>
      </div>
    </div>
  </nav>
  <div
    className="container"
  >
    <div>
      <h1>
        HomeScreen
      </h1>
      <hr />
      <pre>
        {
   "id": 123,
   "name": "Rabi"
}
      </pre>
    </div>
  </div>
</div>
`;
